### PR TITLE
DBZ-2402 Rename some files and update references to those files

### PR DESCRIPTION
**Pull request by TovaCohen**
These updates are required to be able to fetch the upstream content and automatically build the downstream user guide. 
I renamed some files, I updated include statements that reference those files, I updated names of document attributes to replace "CDC" with "Debezium". 

I also fixed two formatting glitches in the filtering.adoc file and the content-based-routing.adoc file. 
I ran Antora locally and the doc built with no errors. 
I spot checked content and it appears to render correctly. 
These changes are needed in the 1.2 branch.

### CODE REVIEW

File: documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc [hunk 1]
```diff
@@ -32,11 +32,11 @@
 
 |`Connected`
 |`boolean`
-|Flag that denotes whether the connector is currently connected to the database server.
+|`Flag that denotes whether the connector is currently connected to the database server.
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc:35]: Could you revert this and the next one below. I had fixed those earlier today.

File: documentation/modules/ROOT/pages/connectors/sqlserver.adoc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[id="debezium-connector-for-sql-server"]
+DebeziumInstall[id="debezium-connector-for-sql-server"]
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/connectors/sqlserver.adoc:1]: This looks a bit odd. Won't this show up in rendered docs?

File: documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc (hunk 1 above)
Comment (TovaCohen) [reply documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-monitoring-streaming-metrics.adoc:35]: Hmmm. I didn't intentionally insert that single quote.  I removed it in this entry and in the next one. Hope that works. This update will be in the commit I'll do in a moment.

File: documentation/modules/ROOT/pages/connectors/sqlserver.adoc (hunk 2 above)
Comment (TovaCohen) [reply documentation/modules/ROOT/pages/connectors/sqlserver.adoc:1]: Yes, sorry! That as unintentional paste and I caught the spot where I meant to paste that. 
Update in the momentary commit.

**Comment by TovaCohen** [comment]
@gunnarmorling  - Thanks for the speedy, careful review! I made those fixes and there's a new commit in this PR.

**Pull request merged**